src/support/hazard.c

```diff
@@ -28,22 +28,18 @@ hazard_grow(WT_SESSION_IMPL *session)
      * Allocate a new, larger hazard pointer array and copy the contents of the original into place.
      */
     size = session->hazard_size;
-    WT_RET(__wt_calloc_def(session, size * 2, &nhazard));
-    memcpy(nhazard, session->hazard, size * sizeof(WT_HAZARD));
+    WT_RET(__wt_calloc_def(session, size * 2, &new_hazard));
+    memcpy(new_hazard, session->hazard, size * sizeof(WT_HAZARD));
 
     /*
      * Swap the new hazard pointer array into place after initialization is complete (initialization
      * must complete before eviction can see the new hazard pointer array), then schedule the
      * original to be freed.
      */
     ohazard = session->hazard;
-    WT_PUBLISH(session->hazard, nhazard);
+    WT_PUBLISH(session->hazard, new_hazard);
 
-    /*
-     * Increase the size of the session's pointer array after swapping it into place (the session's
-     * reference must be updated before eviction can see the new size).
-     */
-    WT_PUBLISH(session->hazard_size, (uint32_t)(size * 2));
```

Comment:
`hazard_size` is only ever used in `hazard_grow` and its parent function `__wt_hazard_set_func` which means there's no interthread communication using the variable. The PUBLISH barrier was actually forcing an ordering between the updated `session->hazard` array pointer and `session->hazard_inuse` so I've moved the barrier there instead.

Reply:
I agree with you on the hazard_size being singly threaded.